api.go

```diff
@@ -0,0 +1,2006 @@
+func (s *CreateBucketInput) SetRetentionPeriod(v int64) *CreateBucketInput {
+	s.RetentionPeriod = &v
+	return s
+}
+
+// SetSSEEnabled sets the SSEEnabled field's value.
+func (s *CreateBucketInput) SetSSEEnabled(v bool) *CreateBucketInput {
+	s.SSEEnabled = &v
+	return s
+}
+
+// SetVPool sets the VPool field's value.
+func (s *CreateBucketInput) SetVPool(v string) *CreateBucketInput {
+	s.VPool = &v
+	return s
+}
+
+type DeleteBucketMetadataSearchInput struct {
+	_ struct{} `type:"structure"`
+
+	// Bucket is a required field
+	Bucket *string `location:"uri" locationName:"Bucket" type:"string" required:"true"`
+}
+
+// String returns the string representation
+func (s DeleteBucketMetadataSearchInput) String() string {
+	return awsutil.Prettify(s)
+}
+
+// GoString returns the string representation
+func (s DeleteBucketMetadataSearchInput) GoString() string {
+	return s.String()
+}
+
+// Validate inspects the fields of the type to determine if they are valid.
+func (s *DeleteBucketMetadataSearchInput) Validate() error {
+	invalidParams := request.ErrInvalidParams{Context: "DeleteBucketMetadataSearchInput"}
+	if s.Bucket == nil {
+		invalidParams.Add(request.NewErrParamRequired("Bucket"))
+	}
+
+	if invalidParams.Len() > 0 {
+		return invalidParams
+	}
+	return nil
+}
+
+// SetBucket sets the Bucket field's value.
+func (s *DeleteBucketMetadataSearchInput) SetBucket(v string) *DeleteBucketMetadataSearchInput {
+	s.Bucket = &v
+	return s
+}
+
+type DeleteBucketMetadataSearchOutput struct {
+	_ struct{} `type:"structure"`
+}
+
+// String returns the string representation
+func (s DeleteBucketMetadataSearchOutput) String() string {
+	return awsutil.Prettify(s)
+}
+
+// GoString returns the string representation
+func (s DeleteBucketMetadataSearchOutput) GoString() string {
+	return s.String()
+}
+
+type EcsIndexableKey struct {
+	_ struct{} `type:"structure"`
+
+	Datatype *string `type:"string"`
+	Name     *string `type:"string"`
+}
+
+// String returns the string representation
+func (s EcsIndexableKey) String() string {
+	return awsutil.Prettify(s)
+}
+
+// GoString returns the string representation
+func (s EcsIndexableKey) GoString() string {
+	return s.String()
+}
+
+// SetDatatype sets the Datatype field's value.
+func (s *EcsIndexableKey) SetDatatype(v string) *EcsIndexableKey {
+	s.Datatype = &v
+	return s
+}
+
+// SetName sets the Name field's value.
+func (s *EcsIndexableKey) SetName(v string) *EcsIndexableKey {
+	s.Name = &v
+	return s
+}
+
+type EcsObjectMatch struct {
+	_ struct{} `type:"structure"`
+
+	IndexKey        *string              `locationName:"indexKey" type:"string"`
+	ObjectId        *string              `locationName:"objectId" type:"string"`
+	ObjectName      *string              `locationName:"objectName" type:"string"`
+	ObjectOwnerZone *string              `locationName:"objectOwnerZone" type:"string"`
+	QueryMds        []*EcsQueryMds `locationName:"queryMds" type:"list" flattened:"true"`
```

Comment:
`QueryMds` is the field name in response body of metadata search, that’s why I used that name. So I’ll leave it as it is if you are ok with that.